### PR TITLE
CUMULUS-2208: Write async operations to ES

### PR DESCRIPTION
**Pull request by markdboyd**
BLOCKED on #2233 and #2245 

**Summary:** Summary of changes

Addresses [CUMULUS-2208: Implement RDS -> ES replication](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2208)

## Changes

* Update logic creating async operations in `async-operations` package to write record directly to ES
* Update logic in async operations Docker image (packages/api/ecs/async-operation) to update record directly in ES
* Update endpoint DELETE logic to delete record directly from ES
* Update unit tests

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [ ] Adhoc testing

### CODE REVIEW

File: packages/api/lib/testUtils.js [hunk 1]
```diff
@@ -214,13 +222,15 @@ function fakeExecutionFactory(status = 'completed', type = 'fakeWorkflow') {
 function fakeAsyncOperationFactory(params = {}) {
   const asyncOperation = {
     taskArn: randomId('arn'),
-    id: randomId('id'),
+    id: uuidv4(),
```
Comment (botanical) [inline packages/api/lib/testUtils.js:225]: nice

File: packages/api/tests/endpoints/test-elasticsearch.js [hunk 2]
```diff
@@ -271,6 +271,7 @@ test.serial('Reindex request returns 400 with the expected message when source i
     index: defaultIndexName,
     body: { mappings },
   });
+  t.teardown(() => esClient.indices.delete({ index: defaultIndexName }));
```
Comment (botanical) [inline packages/api/tests/endpoints/test-elasticsearch.js:274]: Doesn't the test essentially do this on line 276?

File: packages/async-operations/src/async_operations.ts [hunk 3]
```diff
@@ -95,6 +104,53 @@ export const startECSTask = async ({
   }).promise();
 };
 
+export const createAsyncOperation = async (
+  params: {
+    createObject: ApiAsyncOperation,
```
Comment (botanical) [inline packages/async-operations/src/async_operations.ts:109]: Curious to hear why you named this `createObject` 🤔

File: packages/es-client/indexer.js [hunk 4]
```diff
@@ -372,6 +411,33 @@ function deletePdr({
...
+ *
+ * @param  {Object} params
+ * @param  {Object} params.esClient - ElasticSearch Connection object
+ * @param  {string} params.id - the async operation ID
+ * @param  {string[]} [params.ignore] - Array of response codes to ignore
+ * @param  {string} params.index - Elasticsearch index alias (default defined in search.js)
+ * @param  {string} params.type - Elasticsearch type (default: asyncOperation)
+ * @returns {Promise} Elasticsearch response
+ */
+function deleteAsyncOperation({
```
Comment (botanical) [inline packages/es-client/indexer.js:425]: Is there a reason why we don't need to write a unit test for this function?

**Review by botanical — APPROVED** [review]
I pulled this down and tested that an async operation was correctly written to Elasticsearch, DynamoDB, and PostgreSQL and verified that it is deleted in parallel from Elasticsearch, DynamoDB, and PostgreSQL.

File: packages/api/tests/endpoints/test-elasticsearch.js (hunk 2 above)
Comment (markdboyd) [reply packages/api/tests/endpoints/test-elasticsearch.js:274]: Yeah, that's bad duplication. I will remove.

File: packages/async-operations/src/async_operations.ts (hunk 3 above)
Comment (markdboyd) [reply packages/async-operations/src/async_operations.ts:109]: Gives it some delineation between the fact that this the object represents the async operation that we are going to create, but hasn't been created yet until this function returns

File: packages/es-client/indexer.js (hunk 4 above)
Comment (markdboyd) [reply packages/es-client/indexer.js:425]: These functions are just shallow calls directly through to `deleteRecord`, but I'll add some unit tests

**Pull request merged**